tests/test_pono.py

```diff
@@ -0,0 +1,175 @@
+            Ite,
+            solver.make_term(
+                Equal,
+                test_state,
+                solver.make_term(0, bvsort2)
+            ),
+            solver.make_term(1, bvsort2),
+            solver.make_term(
+                Ite,
+                solver.make_term(
+                    Equal,
+                    test_state,
+                    solver.make_term(1, bvsort2)
+                ),
+                solver.make_term(2, bvsort2),
+                solver.make_term(
+                    Ite,
+                    solver.make_term(
+                        Equal,
+                        test_state,
+                        solver.make_term(2, bvsort2)
+                    ),
+                    solver.make_term(3, bvsort2),
+                    solver.make_term(3, bvsort2)
+                )
+            )
+        )
+    )
+
+    # Boolean sort
+    boolsort = solver.make_sort(ss.sortkinds.BOOL)
+
+    # Create statevar to drive `config_en`
+    config_en = rts.named_terms["config_en"]
+    config_en_driver = rts.make_statevar('config_en_driver',
+                                         boolsort)
+    # Starts low
+    rts.constrain_init(solver.make_term(Equal, config_en_driver,
+                                        solver.make_term(False)))
+    # Goes high for a cycle, then low for rest of states
+    rts.assign_next(
+        config_en_driver,
+        solver.make_term(
+            Ite,
+            solver.make_term(
+                Equal,
+                test_state,
+                solver.make_term(0, bvsort2)
+            ),
+            solver.make_term(True),
+            solver.make_term(
+                Ite,
+                solver.make_term(
+                    Equal,
+                    test_state,
+                    solver.make_term(1, bvsort2)
+                ),
+                solver.make_term(False),
+                solver.make_term(
+                    Ite,
+                    solver.make_term(
+                        Equal,
+                        test_state,
+                        solver.make_term(2, bvsort2)
+                    ),
+                    solver.make_term(False),
+                    solver.make_term(False)
+                )
+            )
+        )
+    )
+    # input and state var should be equal
+    rts.add_constraint(
+        solver.make_term(Equal, config_en, config_en_driver)
+    )
+
+    # config_data input is equal to config_data param
+    rts.constrain_inputs(
+        solver.make_term(Equal, rts.named_terms["config_data"],
+                         solver.make_term(config_data, bvsort2))
+    )
+
+    # property imp_res matches spec_res when test_state == 3
+    prop = pono.Property(
+        rts, solver.make_term(Or, solver.make_term(Equal, imp_res, spec_res),
+                              solver.make_term(Distinct, test_state,
+                                               solver.make_term(3, bvsort2))))
+
+    # set solver options
```

Comment:
It'd actually be best to set the solver options above, right when you create the solver.